pipeline/src/aggregator/mod.rs

```diff
@@ -802,10 +833,14 @@ impl Aggregator {
             .await
             .context("writing to mem table data")?;
 
-        let count = self.count_cache().await?;
-        // If we have enough data cached in memory write it out to persistent store
-        if count >= self.max_cached_rows {
+        if allow_cache_flush {
             self.flush_cache().await?;
+            let count = self.count_cache().await?;
+            tracing::debug!(%count, will_flush = %count >= self.max_cached_rows, "counts for mem table");
+            // If we have enough data cached in memory write it out to persistent store
+            if count >= self.max_cached_rows {
+                self.flush_cache().await?;
+            }
```

Comment:
Would be good to document why this check is important. I don't really understand it outside of testing use cases, in particular the block on l301 :thinking: